tests/00-verifications.bats

```diff
@@ -8,3 +8,57 @@ load libs/shared_setup
     run grep -r '^#!.*bin/bash' $BATS_TEST_DIRNAME/..
     assert_output ""
 }
+
+# This test verifies permissions of directories and important files are correct.
+@test "repository file & dir permissions are correct" {
+    # Define executable and non-executable extensions
+    executables=('sh' 'bats')
+    nonexecutables=('bash' 'html' 'jar' 'md' 'out' 'patch' 'php' 'template' 'txt' 'xml' 'xsl' 'yml')
+    # Other nonexecutable files, but looked by name, not by extension
+    otherfiles=('.gitignore' '.htaccess')
+
+    # Directories must be 755
+    run find $BATS_TEST_DIRNAME/.. -type d -not -perm 755 -not -path "*/.git/*"
```

Comment:
Pity, for some reason this is failing in travis, looking for dirs not having 755. Passes here. Will try some alternative solution.